app/locustio/common_utils.py

```diff
@@ -288,4 +288,40 @@ def raise_if_login_failed(locust):
         raise exception.StopUser('Action login_and_view_dashboard failed')
 
 
+def run_as_specific_user(username=None, password=None):
+    if not (username and password):
+        raise SystemExit(f'The credentials are not valid: {{admin: {username}, password: {password}}}.')
+
+    def deco_wrapper(func):
+        @functools.wraps(func)
+        def wrapper(*args, **kwargs):
+            api_url = {
+                'jira': '/rest/api/2/serverInfo',
+                'confluence': '/rest/api/user/anonymous'}
+
+            locust = None
+            for obj in list(locals()['kwargs'].values()) + list(locals()['args']):
+                if isinstance(obj, TaskSet):
+                    locust = obj
+
+            if locust:
+                session_user_name = locust.session_data_storage["username"]
+                session_user_password = locust.session_data_storage["password"]
+                url = api_url[locust.session_data_storage['app']]
+
+                locust.client.cookies.clear()
+                locust.get(url, auth=(username, password), catch_response=True)  # send requests by the specific user
+
+                result = func(*args, **kwargs)
+
+                locust.client.cookies.clear()
+                locust.get(url, auth=(session_user_name, session_user_password), catch_response=True)  # send requests by the session user
+
+                return result
```

Comment:
This return looks redundant. Decorator works without it.

Reply:
Yes, you are right. Need to make improvements in my knowledge of decorators